include/soci/soci-platform.h

```diff
@@ -20,7 +20,7 @@
 #include <cstdlib>
 #include <ctime>
 
-#if defined(_MSC_VER) || defined(__MINGW32__)
+#if defined(_MSC_VER)
```

Comment:
I'm pretty sure that MinGW does use `I64` when using msvcrt.dll, and doesn't support `ll`. So this test must be made conditional on MinGW version/kind, otherwise it will fix some of them while breaking the others.